src/os/bluestore/BlueStore.cc

```diff
@@ -10653,6 +10653,7 @@ int BlueStore::_do_remove(
   txc->removed(o);
   o->extent_map.clear();
   o->onode = bluestore_onode_t();
+  txc->note_modified_object(o);
```

Comment:
txc->removed call above seems to be the only usage for TransContext::removed() func. And the latter (in addition to onodes list update) performs a removal from 'modified_objects' list. Then we insert into it with note_modified_object. May be simply modify removed() function behavior?